pyro/infer/autoguide/guides.py

```diff
@@ -656,44 +657,75 @@ def _loc_scale(self, *args, **kwargs):
         return self.loc, scale
 
 
-class AutoIAFNormal(AutoContinuous):
+class AutoNormalizingFlow(AutoContinuous):
     """
     This implementation of :class:`AutoContinuous` uses a Diagonal Normal
-    distribution transformed via a :class:`~pyro.distributions.transforms.AffineAutoregressive`
-    to construct a guide over the entire latent space. The guide does not depend on the model's
-    ``*args, **kwargs``.
+    distribution transformed via one of the normalizing flows in
+    :mod:`~pyro.distributions.transforms`to construct a guide over the entire
+    latent space. The guide does not depend on the model's ``*args, **kwargs``.
 
     Usage::
 
-        guide = AutoIAFNormal(model, hidden_dim=latent_dim)
+        guide = AutoNormalizingFlow(model, 2, affine_autoregressive)
         svi = SVI(model, guide, ...)
 
     :param callable model: a generative model
-    :param int hidden_dim: number of hidden dimensions in the IAF
+    :param num_flows: number of normalizing flows
+    :param init_transform_fn: a callable which when provided with the latent
+        dimension and additional keyword arguments (`init_transform_kwargs`)
+        returns an instance of
+        :class:`~pyro.distributions.torch_tranform.TransformModule`.
     :param callable init_loc_fn: A per-site initialization function.
         See :ref:`autoguide-initialization` section for available functions.
+    :param init_tranform_kwargs: other keyword arguments taken by
+        `init_transform_fn`.
     """
 
-    def __init__(self, model, hidden_dim=None, init_loc_fn=init_to_median):
-        self.hidden_dim = hidden_dim
-        self.arn = None
+    def __init__(self, model, num_flows, init_transform_fn, init_loc_fn=init_to_median,
```

Comment:
`init_transform_fn` and `init_transform_kwargs` are a bit confusing. The class name is *Flow. We also have the arg `num_flows`. And not all TransformModule can be used here (only transforms have input_dim=output_dim are valid). I would suggest other names: `flow_generator` and `flow_kwargs`. :D

Reply:
I would suggest changing `num_flows` to `num_transforms` and keeping everything in terms of "transforms" rather than "flows". I understand a "normalizing flow" to be comprised of a number of bijective transforms.

Reply:
Done; changed to `transforms` everywhere.